src/Admin/AdminNotices.php

```diff
@@ -301,7 +322,19 @@ protected function is_plugin_scoped_page(): bool {
 
 		$current_page_id = $screen->id;
 
-		return in_array( $current_page_id, $allowed_pages, true );
+		$is_plugin_scoped_page = in_array( $current_page_id, $allowed_pages, true );
+
+		/**
+		 * Filter to determine if the current admin page is within the scope of the plugin's own pages.
+		 * This filter can be used to add additional pages to the list of allowed pages.
+		 *
+		 * The filter receives the following arguments:
+		 *
+		 * @param bool $is_plugin_scoped_page True if the current page is within scope of the plugin's pages.
+		 * @param string $current_page_id The ID of the current admin page.
+		 * @param array<string> $allowed_pages The list of allowed pages.
+		 */
+		return apply_filters( 'graphql_admin_notices_is_plugin_scoped_page', $is_plugin_scoped_page, $current_page_id, $allowed_pages );
```

Comment:
Don't ask me what it _should_ be, but can we think on a more self-documenting name for this? I'm not a fan of `graphql_admin_notices_is_allowed_admin_page`, but at least it doesn't require a definition for what a "plugin-scoped page" means.

Reply:
2 hardest things in programming: 

- naming things
- exactly once delivery
- exactly once delivery

Reply:
Based on the fact that I don't have a better name, we'll run with `graphql_admin_notices_is_allowed_admin_page`